### PR TITLE
make delete!(::IntSet, x) not crash when x == 0

### PR DESCRIPTION
**Pull request by rfourquet**

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
Was there some discussion of this? I'm not entirely sure it's desirable to silently succeed when trying to delete a value that couldn't possibly be put in the structure in the first place.

**Comment by rfourquet** [comment]
I'm not aware of any discussion, but here it only fixes what was clearly a bound-checking mistake, i.e. for all negative numbers it silently succeeds already. Also, `s = Set(1); delete!(s, nothing)` succeeds, but `push!(s, nothing)` fails. So while you concern is valid, it's not limited to `IntSet` and is beyond the scope of this PR.

**Comment by rfourquet** [comment]
But as `eltype(IntSet) == Int`, I would tend to think that it should be valid to `delete!` non-positive `Int` values, so that success/failure depends on the types, not the values. Currently, it's not clarified in the docstring what should happen if the collection has no corresponding key.

**Comment by StefanKarpinski** [comment]
The `eltype` reasoning makes sense to me – +1 to allowing `delete!` on any value of `eltype`.

File: test/intset.jl [hunk 1]
```diff
@@ -139,6 +139,12 @@ end
 
 @testset "pop!, delete!" begin
     s = IntSet(1:2:10)
+    # deleting non-negative values should be no-op
```
Comment (tkelman) [inline test/intset.jl]: non-positive?

Comment (rfourquet) [inline test/intset.jl]: Yes fixed thanks. I updated with a CI-skipped commit -- the original one was all green except for 1 timeout.

**Review by JeffBezanson — APPROVED** [review]

**Pull request merged**